app/models/user.rb

```diff
@@ -963,12 +963,21 @@ def ensure_uniq_email
     raise Exceptions::UnprocessableEntity, 'Email address is already used for other user.'
   end
 
-  def validate_roles
+  def validate_ooo
+    return true if out_of_office != true
+    raise Exceptions::UnprocessableEntity, 'out of office start is required' if out_of_office_start_at.blank?
+    raise Exceptions::UnprocessableEntity, 'out of office end is required' if out_of_office_end_at.blank?
+    raise Exceptions::UnprocessableEntity, 'out of office end is before start' if out_of_office_start_at > out_of_office_end_at
+    raise Exceptions::UnprocessableEntity, 'out of office replacement user is required' if out_of_office_replacement_id.blank?
+    raise Exceptions::UnprocessableEntity, 'out of office no such replacement user' if !User.find_by(id: out_of_office_replacement_id)
+    true
+  end
+
+  def validate_roles(role)
     return true if !role_ids
-    role_ids.each do |role_id|
-      role = Role.lookup(id: role_id)
-      raise "Unable to find role for id #{role_id}" if !role
-      next if !role.preferences[:not]
+    role = Role.lookup(id: role.id)
+    raise "Unable to find role for id #{role_id}" if !role
+    if role.preferences[:not].present?
       role.preferences[:not].each do |local_role_name|
         local_role = Role.lookup(name: local_role_name)
         next if !local_role
```

Comment:
I can't comment on the line below since it's the original state. However we should always have refactoring in mind when working on methods. Therefore please change the line below:

```ruby
raise "Role #{role.name} conflicts with #{local_role.name}" if role_ids.include?(local_role.id)
```

To a guard clause which makes the code more readable:

```ruby
next if role_ids.exclude?(local_role.id)
raise "Role #{role.name} conflicts with #{local_role.name}"
```